src/features/common/Layout/UserPropsContext.tsx

```diff
@@ -98,36 +104,66 @@ export const UserPropsProvider: FC = ({ children }) => {
     logout({ returnTo: returnUrl });
   };
 
+  const fetchUserProfile = useCallback(
+    async (impersonationData?: ImpersonationData) => {
+      const header = {
+        'tenant-key': `${tenantConfig.id}`,
+        'X-SESSION-ID': await getsessionId(),
+        Authorization: `Bearer ${token}`,
+        'x-locale': locale,
+      };
+      try {
+        const response = await fetch(
+          `${process.env.API_ENDPOINT}/app/profile`,
+          {
+            method: 'GET',
+            headers: setHeaderForImpersonation(header, impersonationData),
+          }
+        );
+
+        if (!response.ok) {
+          const errorData = await response.json();
+          throw new APIError(response.status, errorData);
+        }
+        return response.status === 204 ? true : await response.json();
```

Comment:
Why are we handling a 204 response status here? How can that scenario occur while fetching the user profile. 

Even if such a scenario occurs, should we not treat it as an error when user profile data is missing?

Reply:
We usually won't see this scenario for the GET request, however I've handled the error scenario